### ISSUE TITLE
changing scan type crashes GUI

### ISSUE THREAD

**Issue by sbourdeauducq** [m-labs/artiq#333]
[artiq_gui.pyon.zip](https://github.com/m-labs/artiq/files/174161/artiq_gui.pyon.zip)
select "Random scan" and:
``` 
Traceback (most recent call last):
  File "/home/sb/artiq/artiq/gui/scanwidget.py", line 404, in resizeEvent
    self.viewRange()
  File "/home/sb/artiq/artiq/gui/scanwidget.py", line 330, in viewRange
    self.realStop - self.realStart)
TypeError: unsupported operand type(s) for -: 'float' and 'NoneType'
Aborted (core dumped)
```

**Comment by sbourdeauducq** [comment]
Unfolding argument groups with a scan widget in them produces the same crash.

**Comment by jordens** [comment]
fixed in 11bfe33
@sbourdeauducq could you give it a hidpi test?

**Comment by sbourdeauducq** [comment]
Works great!
